### PR TITLE
Only highlight tab on prefix match

### PR DESCRIPTION
**Pull request by philipnrmn**
This PR updates the tab utility so that instead of making a tab active when its name exists anywhere in the route, it only makes the tab active when its name exists at the start of the route. 

- The tab named `universe-packages` will be active with either of the routes `universe-packages` or `universe-packages-detail`.
- The tab named `services` will be active with `services` and `services-detail` but not with `-services-deployment`. 

This PR provides the basis for the fix in #166

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @mlunoe

**Comment by kennyt** [comment]
i remember there was a reason it was implemented this hacky way.

for example, this breaks the highlighting of the package-detail for universe.

px

if you find a way to fix that, then this PR will be cool

**Comment by daiyiwhale** [comment]
Hm ):

**Comment by philipnrmn** [comment]
Thanks for the catch @kennyt.

**Comment by philipnrmn** [comment]
💡 What about if we just use startsWith? That way we can continue to use prefixes in routes, and we can also keep the readability of the `service-deployment` route by introducing a prefix like a hyphen.

File: src/js/utils/__tests__/TabsUtil-test.js [hunk 1]
```diff
@@ -24,22 +25,40 @@ describe('TabsUtil', function () {
...
+    it('should return LIs', function () {
       var result = TabsUtil.getTabs(this.tabs, 'baz', this.getElement);
 
       expect(result[0].type).toEqual('li');
       expect(result[1].type).toEqual('li');
       expect(result[2].type).toEqual('li');
     });
 
-    it('should return element\'s with one active class', function () {
+    it('should return elements with one active class', function () {
```
Comment (philipnrmn) [inline src/js/utils/__tests__/TabsUtil-test.js]: I fixed up a couple of `Deppenapostrophe` while I was in here.

**Comment by philipnrmn** [comment]
Tests and lint are passing locally, looks like the Velocity build failed during `npm install`.

**Comment by pierlo-upitup** [comment]
Yeah, I see a bunch of untar errors like `npm ERR! tar.unpack untar error /root/.npm/core-js/1.2.6/package.tgz`. Weird!

**Comment by pierlo-upitup** [comment]
So how do we get this PR merged @rcorral ? I tried adding `npm cache clean` to our build config but it didn't seem to work. I can't think of other reasons why the build failed here – any tips?

**Comment by rcorral** [comment]
We run the tests again :) trying that now.

File: src/js/utils/__tests__/TabsUtil-test.js (hunk 1 above)
Comment (kennyt) [inline src/js/utils/__tests__/TabsUtil-test.js]: good stuff

File: src/js/utils/__tests__/TabsUtil-test.js [hunk 2]
```diff
@@ -24,22 +25,42 @@ describe('TabsUtil', function () {
...
       var result = TabsUtil.getTabs(this.tabs, 'baz', this.getElement);
 
       expect(result[0].props.className).toEqual('tab-item');
       expect(result[1].props.className).toEqual('tab-item active');
       expect(result[2].props.className).toEqual('tab-item');
     });
 
+    it('should not highlight routes contained within the class name',
+        function () {
+          var tabs = TabsUtil.getTabs(this.hierarchicalTabs, '-foo',
```
Comment (mlunoe) [inline src/js/utils/__tests__/TabsUtil-test.js]: See my note here: https://github.com/dcos/dcos-ui/pull/166/files#r63264866

I think we should update this test to not use the `-` prefix

**Pull request merged**